### DISCUSSION TITLE
multiple editable windows

### DISCUSSION THREAD

**Discussion by alexisargyris** [vkbo/novelWriter#721] [Ideas]
Hello,

Often, I find myself needing to edit two files at the same time. On such occasions I switch to VS Code and then return back to novelWriter. Even that, though, is not very workable because of the file names. I end up touching the two files I need in novelWriter and then switching to VS Code where I list the files by last access so those I'm interested in come up at the top.

Any plan for allowing such a use case?

Best regards

**Comment by vkbo** [comment]
Yes, editing the files externally isn't really ideal even if it technically works. The `ToC.txt` file contains the link between the hash file name and the internal document label.

In an earlier iteration of novelWriter, I used editor tabs. The current iteration instead is modelled after the user interface of zim-wiki which can only edit one document at a time. The entire user interface is built around the single editor class, so rewriting this is a pretty substantial task.

The decision to not go in the multiple editor direction was deliberate. It's also how other similar programs work, so I concluded this would be sufficient and even cleaner and less cluttered. I still opted to make it fairly easy to switch between documents to edit. Since you can only actually edit one document at a time even with tabs, I think it would be a better approach to make it easier to switch between active documents. In zim-wiki there is the option to keep a list of recent edited pages above the editor as buttons. There is practically no difference between moving in the edit history and switching tabs.

**Comment by alexisargyris** [comment]
Thanks for the toc.txt info, this will help. What about running two instances of novelWriter on the same project? Would it be easier from an implementation point of view?

**Reply by vkbo** [reply]
novelWriter holds the entire project meta structure in memory while editing a project, so that would not work. That's why there is a lock file in place when the project is open. You can technically do it, at your own risk, by overriding the lock and refrain from changing the project structure while the other editor is open. The document file content is completely independent, so that's fine.

Do you need to edit two files side by side? That's something I've never needed myself, even when programming, so I guess it hasn't really occurred to me. Viewing a file for reference is generally what I need. If you're just looking for a quick way to switch between files in edit mode, then it's a much simpler thing to implement as I mentioned above. Would that be sufficient?

**Reply by alexisargyris** [reply]
Just being able to reference another file would be very helpful indeed.

For me at least, the point of having multiple files open comes relatively late in the writing process when I need to ensure that two different scenes talk in a consistent manner about some thing. If one has good memory then this need shouldn't arise as much, I suppose.

**Reply by vkbo** [reply]
Ah, I see. That does make sense.

I do something similar early on too when I outline the plot. I build my plot using a process similar to the snowflake method where I grow it in steps while keeping earlier steps in other files. I generally keep one file in view mode when I edit the next stage. If I do need to go back and edit older files for consistency, I have them next to each other in the project tree anyway.

I do plan to add filtering options to the left side tree panel to be able to for instance list all pieces of text where a specific character or other reference is tagged. It would also make it easier to look through scenes to check for consistency.

I'll keep your use case in mind anyway. I think there are a few possible solutions.
